### ISSUE TITLE
Sea Animals Detection using Deep Neural Networks

### ISSUE THREAD

**Issue by PiyushBL45t** [abhisheks008/DL-Simplified#349]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Sea Animals Detection using Deep Learning</br>
:red_circle: **Aim** : To create a DL Model that will detect the sea animals and creatures</br> 
:red_circle: **Dataset** : https://www.kaggle.com/datasets/vencerlanz09/sea-animals-image-dataste </br> 
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Piyush Bhujbal
- GitHub Profile Link : https://www.github.com/PiyushBL45t
- Email ID : bhujbalpiyush@gmail.com
- Participant ID (if applicable):
- Approach for this Project :
        1. Perform Image preprocessing if required.
        2. Then perform image visualization.
        3. Create training, testing and validation data sets using image data generation.
        4. Create a sample summary of the CNN model.
        5. Use a proper optimizer to construct CNN and optimize the algorithm.
        6. Implement model training using CNN and pre-trained models.
        7. Evaluate each models performance.
        8. Make predictions using test dataset.
        9. Document all the parameters like epochs, training time as well as accuracy and loss functions used.
- What is your participant role? (Mention the Open Source program): Social Summer of Code Participant

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
Issue assigned to you @PiyushBL45t

**Comment by PiyushBL45t** [comment]
Thank you.

**Comment by abhisheks008** [comment]
Is this the one you are talking about?
@PiyushBL45t

**Comment by PiyushBL45t** [comment]
Yes

**Comment by abhisheks008** [comment]
You can edit this issue template and start working on it. Make sure you complete it within the deadline.

**Comment by abhisheks008** [comment]
Can you clarify more about it?

**Comment by Surya735** [comment]
I would like to work on this issue. Can you please clarify what 'detection'  encompasses? Does it mean to 'classify sea animals' ?

**Comment by abhisheks008** [comment]
Yes. Can you please share the approach for solving this issue/project?
@Surya735

**Comment by Surya735** [comment]
Hi @abhisheks008, 
My initial approach would be using self supervised learning for classification and would consist of the following:

1) A convolutional network to learn features, constrained to allow the features to project data points onto lower dimensional manifolds

2) A second network that focuses on classification of data points on the basis of the manifolds on which the feature vectors obtained from above lie

**Comment by abhisheks008** [comment]
Hi @Surya735 it is a nice approach for solving this issue. Try to implement 3-4 deep learning methods and make a comparison to find out the best fitted model based on the accuracy scores of the models generated. You can check out the existing projects in this repository to get an idea about the same.

Issue assigned to you under OSEN. Go ahead!

**Comment by Surya735** [comment]
Thank you for assigning the issue to me @abhisheks008

**Comment by abhisheks008** [comment]
Hi @Surya735, updates on this issue?

**Comment by Surya735** [comment]
Hey @abhisheks008 ,

I have finished the necessary literature review and will begin working on the code now.
Although I have my end semester exams starting shortly prior to which I have a round of mini project submissions for various subjects. So unfortunately I won't have a lot of time to work on it currently and I really hope that's not an issue because I would genuinely love to contribute to the issue. Are you looking at any deadline on this?

**Comment by abhisheks008** [comment]
Hi @Surya735, there is no such deadline from my side. But as you are a part of a particular program, I believe following the deadline of OSEN comes under code of conduct, which you should follow.

Anyways take your time and all the best for your semester exams. Good luck!

**Comment by abhisheks008** [comment]
This issue will not be considered under **CollabCode** Open Source event organized by **OSEN** as the program ended on **November 19th, 2023**.

**Comment by aindree-2005** [comment]
Full name : Aindree Chatterjee
GitHub Profile Link : https://www.github.com/aindree-2005
Email ID : [aindree2005@gmail.com](mailto:aindree2005@gmail.com)
Approach for this Project : Using CNNS and data augmentation
What is your participant role? CodePeak
@abhisheks008 can I work on this?
